Fbx importer improvements

alsed opened the pull request:
_Edit: ~~Moved this PR into a draft~~. Most of the changes done here could also be incorporated into the Assimp importer (with the exception of pre and post-rotation). 
The axis correction should also be incorporated into the import root motion, which is done somewhere else. 
I'm unsure wether keep adding more things until this current proposal is reviewed._ 


- Correct import of FBX files with different coordinate systems (e.g., Z up).
- Correct conversion of FBX meshes with geometric transformation.
- Correct the orientation of the viewport camera. Now preview and scene cameras are matched.
- ~~Prefab is now the default for import models.~~
- An Import Animation option `Bake Rotations` is added. If enabled, pre and post rotation data will be added into the keyframes animations. This can correct the import of animations assets that don't come with prebaked rotations (e.g. Unreal assets).


Before:
<img width="907" height="620" alt="Captura desde 2025-08-29 12-30-36" src="https://github.com/user-attachments/assets/4be49f4c-b8f9-4a50-b36e-4912f5ea70eb" />


After:
<img width="902" height="618" alt="Captura desde 2025-08-30 20-45-02" src="https://github.com/user-attachments/assets/060dcffd-1d6c-49fb-bc2a-1d18cd7cf54a" />

alsed committed:
Improvements to the fbx importer

- FBX files with an Up axis different from the Y axis are now rotated and
positioned correctly in the engine.
- Geometric transformation has been incorporated for meshes, and the copying
of this data has been corrected.
- Fixed a bug where the importer did not correctly detect the axes of the
FBX file.
    “UpAxi used the UpVector enum (from ofbx), but for some reason it
works better to use the integer values directly.
    “GlobalSettings” is now created from “OpenFbxImporter” and called from
there.
- “ConvertRootRotation” has been replaced by a "ToFlax" matrix
used to correctly rotate and position meshes, not at the root level.
- The main functions of “OpenFbxImporter” have been moved outside. This
struct has been left as an auxiliary data type.
- Two new structs, “AxisSystem” and “CoordSystem,” have been created,
which are created in “OpenFbxImporter.”
The parsing of the coordinate system has been moved to its own function.
-A function inside "CoordSystem" has been created to evaluate the handedness
of the FBX file.
- The Scene object destructor is now called.

alsed committed:
reposition viewport camera

alsed committed:
set prefab as default for model import

alsed committed:
Fix orientation of viewport cameras

alsed committed:
Removes unused code

Tryibion commented:
My opinion: I don't use the prefab option that often... so I would prefer it to not be the default.

xxSeys1 commented:
> My opinion: I don't use the prefab option that often... so I would prefer it to not be the default.

Maybe another thing to put in a "Defaults" category in the editor settings? Like in #3296

alsed committed:
fix camera-arcballview for Assets Preview

alsed committed:
restores use of const value arcBallOrientation for Asset Preview Camera

alsed committed:
2nd attempt to fix compilation error

alsed committed:
Revert "set prefab as default for model import"

This reverts commit 4bde6fea2f6f02df45f583c873c0d4bb69366399.

alsed commented:
Additional description to the changes.

Changes to the FBX importer:
- Most of the functions inside the struct "OpenFbxImporterData" where moved outside.
- The code to parse the fbx system coordinate was also moved to its own function and a additional struct was created to ease the process ("Axis System").
- Also, a struct "CoordSystem" was created to interpret the system.
- The quaternion "RootConvertTransform" was replaced by a transformation matrix "To Flax" which is constructed inside "CoordSystem".

Changes to the preview cameras:
- The magic values that controlled the camera orientation were centralized in private values "ViewPort Orientation" and "ArcBallOrientation".
- "ArcBallOrientation" was reoriented and expressed in Euler angles.

Notes:
- prefab as default has been reverted as suggested by @Tryibion.
- In case the previewlight needs to be reoriented, this can be done in `AssetsPreview` file.

alsed committed:
Add pre and post rotation transformations to animations

alsed committed:
Fix translation in animations with preRotation

alsed committed:
Add Import Animation Option "Bake Transforms"

- Add the option to bake pre and post rotations into the animation keyframes for FBX
- Add SystemCoord transformation for animations
- Some changes in the comments for clarity

alsed commented:
**Update:**

An Import Animation option `Bake Rotations` is added:
If enabled, pre and post rotation data will be calculated into the keyframes animations.
This can correct the import of animations assets that don't come with prebaked rotations (e.g. Unreal assets).

<img width="777" height="460" alt="Bake Rotations option" src="https://github.com/user-attachments/assets/94223e9f-7ae1-4bb1-917e-f5d6d570459c" />

Before:
<img width="466" height="527" alt="bef" src="https://github.com/user-attachments/assets/4438a392-54ac-462b-a905-b3346a3ca7ad" />

After:
<img width="466" height="527" alt="after" src="https://github.com/user-attachments/assets/7d3b93ee-310c-4302-b994-eaec9c5f1a27" />
